cmd/drone-docker-ecr/main.go

```diff
@@ -32,14 +32,16 @@ func main() {
 		region = defaultRegion
 	}
 
-	os.Setenv("AWS_ACCESS_KEY_ID", key)
-	os.Setenv("AWS_SECRET_ACCESS_KEY", secret)
 	os.Setenv("AWS_REGION", region)
+	awsConfig := aws.Config{Region: &region}
 
-	sess, err := session.NewSession(&aws.Config{
-		Credentials: credentials.NewEnvCredentials(),
-		Region:      &region,
-	})
+	if key != "" && secret != "" {
+		os.Setenv("AWS_ACCESS_KEY_ID", key)
```

Comment:
I had thought that the AWS SDK had built-in understanding of these environment variables if you use the default credential chain? Are we sure that we even need to be handling them at all?

Reply:
I removed the extra credential provider. You are right the default chain will use them when available.

Reply:
You're right, I removed the explicit credential provider and allow the default chain to find the correct credentials.